app/components/login_button_component.rb

```diff
@@ -0,0 +1,60 @@
+class LoginButtonComponent < BaseComponent
+  attr_reader :action, :big, :color, :width, :height, :logo_path, :tag_options
+
+  alias_method :big?, :big
+
+  DEFAULT_HEIGHT = 16
+  DEFAULT_WIDTH = 121
+  BIG_HEIGHT = 20
+  BIG_WIDTH = 152
+
+  def initialize(
+    action: ->(**tag_options, &block) { button_tag(**tag_options, &block) },
+    big: false,
+    color: "light blue",
+    **tag_options
+  )
+    @action = action
+    @big = big
+    @color = color
+    @tag_options = tag_options
+  end
+
+  def logo_path
+    return "logo-white.svg" if color == "dark blue"
+    "logo.svg"
+  end
+
+  def width
+    return BIG_WIDTH if big?
+    DEFAULT_WIDTH
+  end
+
+  def height
+    return BIG_HEIGHT if big?
+    DEFAULT_HEIGHT
+  end
+
+  def css_class
+    classes = ['usa-button', *tag_options[:class]]
+    classes << 'usa-button--big' if big
+    classes << 'bg-white text-primary-darker border border-base hover:border hover:bg-white hover:text-primary-darker hover:border-base' if color == "white"
+    classes << 'bg-primary-lighter text-primary-darker hover:bg-primary-lighter hover:text-primary-darker' if color == "light blue"
+    classes << 'bg-primary-darker text-white hover:bg-primary-darker hover:text-white' if color == "dark blue"
```

Comment:
Rather than using utility classes, it might be easier and more maintainable to create a component-specific stylesheet, where each color variation could be defined as a [BEM modifier](https://en.bem.info/methodology/naming-convention/#two-dashes-style).

e.g.

```scss
// login_button_component.scss

.login-button.login-button--light-blue {
  @include u-bg('primary-lighter');
  // ...
}
```